src/sentry/consumers/__init__.py

```diff
@@ -396,12 +400,34 @@ def build_consumer_config(group_id: str):
             healthcheck_file_path, strategy_factory
         )
 
+    if dlq_topic is not None:
+        try:
+            cluster_setting = get_topic_definition(dlq_topic)
+        except ValueError:
+            click.BadParameter(
+                "Invalid DLQ topic: {dlq_topic}, all DLQ topics must have topic definition configured in settings"
+            )
+
+        producer_config = get_kafka_producer_cluster_options(cluster_setting)
+        producer_config.pop("compression.type", None)
+        producer_config.pop("message.max.bytes", None)
+        dlq_producer = KafkaProducer(producer_config)
+
+        dlq_policy = DlqPolicy(
+            KafkaDlqProducer(dlq_producer, Topic(dlq_topic)),
+            DlqLimit(),
```

Comment:
I think we should provide a limit here. Otherwise it's possible 100% of messages go to DLQ

Reply:
Sounds good, what should a sensible limit be?

Reply:
Not sure. Depends on the use case. This is what we picked Snuba wide: https://github.com/getsentry/snuba/blob/749ef98fe9a0dd319e68ca4f9ca8cba5a2f54c8f/snuba/consumers/consumer_builder.py#L382-L385. I think it'd be ok to start with the same.

Alternatively, if we think the same really doesn't apply across all consumers we could make it a CLI arg. Though no one else has DLQ now, so either way is fine for a first iteration.